src/classes/RLLP_OppPartialSoftCreditRollup.cls

```diff
@@ -64,14 +64,20 @@ public class RLLP_OppPartialSoftCreditRollup {
         set<id> rectypeset = RLLP_OppRollup_UTIL.recordTypesToExcludeCons;
         string ocrQuery = RLLP_OppRollup_UTIL.buildSoftCreditOCRQuery();
         string pscQuery = RLLP_OppRollup_UTIL.buildSoftCreditPSCQuery();
-        Set<Id> primaryOCRs = new Set<Id>();
-        
+        Set<Id> hardCreditOCRs = new Set<Id>();
         Map<Id, List<OpportunityContactRole>> opportunityContactRoles = new Map<Id, List<OpportunityContactRole>>();
         for (OpportunityContactRole ocr : database.query(ocrQuery)) {
-            if (ocr.isPrimary) // keep track of primary OCR to exclude from SC rollup.
-                primaryOCRs.add(ocr.Id);
-            if (!opportunityContactRoles.containsKey(ocr.ContactId)) 
+            if (UTIL_CustomSettingsFacade.getHouseholdsSettings().npo02__Always_Rollup_to_Primary_Contact__c == false &&
```

Comment:
I think it would be great to refactor this logic into a method: `private Boolean isHardCredit(OpportunityContactRole ocr)`. What do you think?

Reply:
I like it! It will be more readable. I will refactor and let you know.